common/lib/client/auth.js

```diff
@@ -260,7 +259,13 @@ var Auth = (function() {
 						}
 					}
 					cb(null, body);
-				});
+				};
+				Logger.logAction(Logger.LOG_MICRO, 'Auth.requestToken().tokenRequestCallback', 'Sending; ' + authOptions.authUrl + '; Params: ' + JSON.stringify(authParams));
+				if(authOptions.authMethod && authOptions.authMethod.toLowerCase() === 'post') {
+					Http.postUri(rest, authOptions.authUrl, authHeaders || {}, authParams, {}, authUrlRequestCallback);
```

Comment:
My argument is that the client may require params and/or headers to support its general processing/validation/routing of the URL. Processing of the body will always be specific to this particular function (and will have access to params in any case) whereas processing of the URL for routing or validation is much more likely to be generic.

> if you wanted to add query params to the URL you could simply change the URL

That's not what's implemented.